PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzurePlugin.java

```diff
@@ -1,438 +0,0 @@
-import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.io.FileUtils;
-import org.apache.commons.lang3.StringUtils;
-import rx.Observable;
-import rx.schedulers.Schedulers;
-
-import javax.annotation.Nonnull;
-import javax.swing.event.EventListenerList;
-import java.io.BufferedOutputStream;
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.URL;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Objects;
-import java.util.Set;
-import java.util.concurrent.TimeUnit;
-import java.util.zip.ZipEntry;
-import java.util.zip.ZipInputStream;
-
-import static com.microsoft.azure.toolkit.ide.common.store.AzureConfigInitializer.TELEMETRY;
-import static com.microsoft.azure.toolkit.ide.common.store.AzureConfigInitializer.TELEMETRY_PLUGIN_VERSION;
-import static com.microsoft.azure.toolkit.intellij.common.AzureBundle.message;
-import static com.microsoft.azuretools.telemetry.TelemetryConstants.*;
-
-@Slf4j
-public class AzurePlugin implements ProjectActivity, DumbAware {
-    public static final String PLUGIN_VERSION = com.microsoft.azure.toolkit.intellij.common.CommonConst.PLUGIN_VERSION;
-    public static final String AZURE_LIBRARIES_VERSION = "1.0.0";
-    public static final String JDBC_LIBRARIES_VERSION = "9.4.0.jre8";
-    public static final int REST_SERVICE_MAX_RETRY_COUNT = 7;
-    private static PluginStateListener pluginStateListener = null;
-    private static final int POP_UP_DELAY = 30;
-
-    // User-agent header for Azure SDK calls
-    public static final String USER_AGENT = "Azure Toolkit for IntelliJ, v%s, machineid:%s";
-
-    public static final boolean IS_WINDOWS = SystemInfo.isWindows;
-
-    public static final boolean IS_ANDROID_STUDIO = "AndroidStudio".equals(PlatformUtils.getPlatformPrefix());
-
-    public static String pluginFolder = PluginUtil.getPluginRootDirectory();
-
-    private static final EventListenerList DEPLOYMENT_EVENT_LISTENERS = new EventListenerList();
-
-    public static List<DeploymentEventListener> depEveList = new ArrayList<DeploymentEventListener>();
-
-    private String installationID;
-
-    private Boolean firstInstallationByVersion;
-
-    @Override
-    public Object execute(@Nonnull Project project, @Nonnull Continuation<? super Unit> continuation) {
-        initializeAIRegistry(project);
-        // Showing dialog needs to be run in UI thread
-        initializeWhatsNew(project);
-        if (!IS_ANDROID_STUDIO) {
-            log.info("Starting Azure Plugin");
-            firstInstallationByVersion = isFirstInstallationByVersion();
-            try {
-                //this code is for copying componentset.xml in plugins folder
-                copyPluginComponents();
-                if (pluginStateListener == null) {
-                    pluginStateListener = new PluginStateListener() {
```

Comment:
may need to add the feature of PluginStateListener to track the installation/uninstall